pkg/errors/errors.go

```diff
@@ -8,59 +8,170 @@ import (
 	"strings"
 
 	perrors "github.com/pkg/errors"
+	"go.uber.org/multierr"
 	"google.golang.org/grpc/codes"
 	status "google.golang.org/grpc/status"
+
+	moccodes "github.com/microsoft/moc/pkg/errors/codes"
+	"github.com/microsoft/moc/rpc/common"
 )
 
+// MocError is a trivial implementation of error that wraps the moc rpc Error struct.
+type MocError struct {
+	// None of MocError's fields should be modifiable after instantiation
+	// since errors are supposed to be immutable.
+	err *common.Error
```

Comment:
+1, I also would like to see the use cases of these functions.

Reply:
Yeah, the divide is basically "default moc error code error" and "translating last error into a moc error code error without losing information".

So NewMocError for example helps create the base error codes (e.g., Not Found and its backwards-compatible message).

While NewMocErrorWithError is used to translate the last error struct across agents. By the time an error reaches cloudagent, it may have multiple layers, e.g., Msg: "The container 'x' was missing in location 'y': Not Found", Code:2. So we use NewMocErrorWithError to retain the message while recreating the moc error.

(Example of NewMocErrorWithError: https://github.com/microsoft/wssdcloudagent/blob/08bd4b8732526eec308e705f263f3c5bdd73add7/pkg/cloudagent/services/storage/virtualharddisk/wssd/wssd.go#L288)